### ISSUE TITLE
Receiving focus is A, but removing is AAA

### ISSUE THREAD

**Issue by erikkroes** [w3c/wcag#1782]
A user tries to buy a subscription that has 3 levels of features and accordingly, price. The different subscriptions can be selected through a radiogroup. The user navigates the page using the tab-key and tabs to te radiogroup
If the user focusses the radiogroup and a new window opens to advertise the subscriptions, that's a failure of 3.2.1. No level A for you!
https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html

The webpage has an extraordinary developer working. They've read WCAG. Instead of showing the window on receiving focus, they change the code slightly so the window shows when the user tabs away from the radiogroup. Not "onFocus" but "onBlur". That's also a failure, but not for 3.2.1. It's a failure for 3.2.5. Suddenly you can get AA where you couldn't even hit A before.
https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F9

Is this intentional? Is there such a big difference between "onBlur" and "onFocus"?

My suggestion would be to **change 3.2.1 to cover all changes in focus**.

**Comment by patrickhlauke** [comment]
Arguably already a more general failure of 2.4.3 Focus Order https://www.w3.org/TR/WCAG/#focus-order

**Comment by alastc** [comment]
It's from before my time, but I suspect that either:
- There are some valid use-cases for changing context onBlur, and/or
- onBlur wasn't seen as a common issue, and/or
- It could be covered by focus-order, as it is not following the order of the page.

If there were a good argument to cover onblur (i.e. it happens in the wild and doesn't have valid use-cases) we'd need to add a new SC to cover that in a post WCAG 2.2 version.

**Comment by patrickhlauke** [comment]
Casting my mind back, I believe the reason for the SC covering the `onfocus` type scenario specifically was an abundance of "solutions" around at the time to prevent unsightly focus outlines on things by immediately blurring them. See for instance https://www.w3.org/WAI/WCAG21/Techniques/failures/F55

**Comment by patrickhlauke** [comment]
And yes, arguably again (with my take on 2.4.3) failures of 3.2.1 could also be seen as special cases of 2.4.3, so there's definitely some overlap. Maybe post 2.2 this overlap could be resolved by deprecating 3.2.1 and expanding the understanding/purpose of 2.4.3 ...

**Comment by JAWS-test** [comment]
>  3.2.1 could also be seen as special cases of 2.4.3,

I think that would not be correct, because 2.4.3 is only about focus order, but 3.2.1 is about context changes. Context changes, by their definition, do not have to be focus changes. If we remove 3.2.1 (and 3.2.2 right along with it?), then all context changes are allowed that do not affect focus order

> Changes in context include changes of:
> 1.    user agent;
>  2. viewport;
>  3. focus;
>  4. content that changes the meaning of the Web page

**Comment by patrickhlauke** [comment]
fair enough (though out of those 4 listed, only the last one isn't a change of focus per, as changing user agent and viewport also result in moving focus - to the new user agent or viewport)

**Comment by JAWS-test** [comment]
> as changing user agent and viewport also result in moving focus - to the new user agent or viewport

@patrickhlauke In my opinion, viewport means the visible page area. This can change without changing the focus. So the page can be scrolled, but the focus remains. This would be a violation of 3.2.1, but not a violation of 2.4.3.

**Comment by JAWS-test** [comment]
@erikkroes In my opinion, onBlur is also covered by 3.2.1. Because the SC says nothing about the technical implementation, but only about the fact that something happens at focus. An onBlur means nothing else than that an element loses the focus because a new one gets the focus. Thus, an onBlur is identical to the onFocus of another element. 3.2.1 can also be violated without onBlur and onFocus, e.g. with an onKeyup event. This is triggered as soon as the tab key is released, with which the element receives the focus.

**Comment by mraccess77** [comment]
Changing context onblur has been seen as acceptable by many.  For example, say you make a choice with radio buttons using the arrows and then tab away from the group.   Based on your select onblur could be used to change the contents of the page below the radio button group.  This is historically been seen as ok by many as long as the content change comes after in the focus order.  When a control issues onblur it doesn't mean anything is automatically focused although maybe the document itself.

**Comment by JAWS-test** [comment]
>  Based on your select onblur could be used to change the contents of the page below the radio button group. 

I think that is also allowed with onfocus and oninput (3.2.2) because it is not a context change by the definition.

**Comment by erikkroes** [comment]
@JAWS-test How do you reason that 3.2.1 covers onBlur? It very explicitly states "receives focus"

@mraccess77 Your story feels like an exception on an exception (radio group and a change further in the DOM) and quite anecdotal. I'm not sure how it helps clarify the difference.

**Comment by mraccess77** [comment]
My point was that a change on focus is very different than most changes on blur.  Changes onFocus are more likely to be problematic and prevent access than use of  onBlur.  OnBlur could be used to break something but it would likely also fail another criteria as well such as focus order.

**Comment by JAWS-test** [comment]
@erikkroes 

WCAG is technology neutral. As I wrote earlier, it doesn't matter what technology is used to create an onfocus event. With JavaScript there are the following possibilities:

```
<input aria-label=1>
<input aria-label=2 onfocus>


<input aria-label=1 onblur>
<input aria-label=2>


<input aria-label=1 onkeydown>
<input aria-label=2>


<input aria-label=1>
<input aria-label=2 onkeyup>


<input aria-label=1>
<input aria-label=2 onmousedown>


<input aria-label=1>
<input aria-label=2 onmouseup>


<input aria-label=1>
<input aria-label=2 onclick>
```

All these examples cause an onfocus violation against SC 3.2.1 to occur at the second field when navigating from the first to the second input field.

**Comment by alastc** [comment]
> My point was that a change on focus is very different than most changes on blur.

I think this is key. To progress, someone would need to do some due-diligence and gather examples of where onblur is used negatively, and non-negatively.

If there are positive uses-cases for changing things onblur, or even just neutral ones, it wouldn't be suitable to include as a guideline.

**Comment by patrickhlauke** [comment]
off the top of my head, purely for a negative example: a form control that, when blurred, is checked for validity, and if found invalid focus is forcibly placed back on the control (creating a temporary focus trap, in essence, until the control is valid). not super common, but i do recall stumbling across these sorts of things on rare occasions.
